ballista/rust/core/src/serde/logical_plan/from_proto.rs

```diff
@@ -157,17 +157,17 @@ impl TryInto<LogicalPlan> for &protobuf::LogicalPlanNode {
                         Some(r?)
                     }
                 };
-
```

Comment:
the partition count is still hard coded here, perhaps it needs to be serialized and deserialized as a field in `LogicalPlanType::ParquetScan` ?

Reply:
Perhaps we should move `max_partitions` out of the logical plan nodes and postpone it to physical plan generation. In `TableProvider.scan` for example.

Reply:
How do we get the value for `max_partitions` in the scan method? Right now this config is provided by the user when creating the TableProvider struct, so I think something needs to be changed there.

Reply:
Agree, I think `max_partitions` shouldn't be bind to a table (in logical plan), but let the planner pass in desired execution parallelism during physical planning, and partition the table files accordingly then.

Reply:
I agree, it sounds like this could benefit from https://github.com/apache/arrow-datafusion/issues/64 too.

Reply:
Okay, I'm going to do this after I've combed through the code.